modules/launcher/src/main/java/org/mule/module/launcher/log4j2/LoggerContextCache.java

```diff
@@ -121,15 +126,45 @@ LoggerContext getLoggerContext(final ClassLoader classLoader)
         final LoggerContext ctx;
         try
         {
-            final int key = computeKey(classLoader);
-            ctx = activeContexts.get(key, new Callable<LoggerContext>()
+            final Integer key = computeKey(classLoader);
```

Comment:
Just so the wrapping is done at this point only and not each time it is used in call to a Map method.